source/index.js

```diff
@@ -147,7 +174,14 @@ module.exports = async (input = 'patch', options) => {
 						return `Private package: not publishing to ${pkgManagerName}.`;
 					}
 				},
-				task: (context, task) => publishTaskHelper(pkgManager, task, options, input)
+				task: async (context, task) => {
+					try {
+						await publishTaskHelper(pkgManager, task, options, input);
```

Comment:
`publishTaskHelper` returns an Observable, not a Promise.

Reply:
Would this work instead:

```js
return publishTaskHelper(pkgManager, task, options, input)
							.subscribe(null, rollback, () => {
								isPublished = true;
							});
```

I'm not very familiar with Observables, but looking at the docs it seems like this should work.

Reply:
@sindresorhus Ping 🙃

Reply:
That doesn't work. The observable needs to be returned in the function because otherwise it will never render the one-time password input...

What you probably want is something like this

```ts
return publishTaskHelper(pkgManager, task, options, input).pipe(
    catchError(() => {
        // catchError expects a new observable/promise to be returned.
 
		return rollback();
    })
);
```